relay/shfu_grpc/client.go

```diff
@@ -0,0 +1,212 @@
+package shfu_grpc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/ibc-go/v8/modules/core/exported"
+	"github.com/datachainlab/lcp-go/relay/shfu_logger"
+	"github.com/datachainlab/lcp-go/relay/shfu_storage"
+	"github.com/google/uuid"
+	"github.com/hyperledger-labs/yui-relayer/core"
+	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+// GetLatestFinalizedHeader retrieves the latest finalized header from gRPC server
+func GetLatestFinalizedHeader(ctx context.Context, grpcAddress string, chainID string, counterpartyChainID string, codec codec.ProtoCodecMarshaler) (core.Header, error) {
+	logger := shfu_logger.GetSHFULogger(ctx)
+	requestID := uuid.Must(uuid.NewV7()).String()
```

Comment:
The following change would make this function cleaner because it can remove more than 30 lines:

```suggestion
	requestID := uuid.Must(uuid.NewV7()).String()
	logger := shfu_logger.GetSHFULogger(ctx).With(
		"request_id", requestID,
		"grpc_address", grpcAddress,
		"chain_id", chainID,
		"counterparty_chain_id", counterpartyChainID,
	)
```

The same applies to many functions.

Reply:
fixed